core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -290,7 +298,27 @@ public Snapshot apply() {
         operation(),
         summary(base),
         base.currentSchemaId(),
-        manifestList.location());
+        manifestList.location(),
+        lastRowId,
+        addedRows);
+  }
+
+  private Long calculateAddedRows(List<ManifestFile> manifests) {
+    return manifests.stream()
+        .filter(
+            manifest ->
+                manifest.snapshotId() == null
+                    || Objects.equals(manifest.snapshotId(), this.snapshotId))
```

Comment:
should we use `snapshotId()` method (instead of `this.snapshotId`)? 

there is no practical difference, since earlier call of `snapshotId()` already set it. but it is probably better for safety to avoid such implicit code ordering dependency